### DISCUSSION TITLE
Where to host Prisma / SQL in the cloud?

### DISCUSSION THREAD

**Discussion by dhuber666** [prisma/prisma#4758] [Q&A] [answered]
Hi guys,

what is the current best hoster for SQL Database that is also compatible with Prisma? For example Heroku won't work because of #4109 or #4751 

I want to get my feet wet with Prisma but right now I don't want to run a local DB. So any suggestions for hosting providers for SQL that are compatible with this shadow db bug?

Thx Geralt

**Comment by ryands17** [comment, marked as answer]
Hey @dhuber666 👋 

One of the options that you could choose [AWS RDS](https://aws.amazon.com/rds/).

Creating a single database will be free for the first 12 months from your account creation and will be paid thereafter.

**Reply by dhuber666** [reply]
Hi @ryands17 

I tried it with a RDS PostgreSQL. After creation I used this `DATABASE_URL` : `postgres://postgres:mypassword@myamazonurl`

But after I want to migrate I get this error: `Error: P1001: Can't reach database server at `myamazonurl":"5432"`

**Reply by ryands17** [reply]
@dhuber666 It seems you have not provided public access to the database and configured your security group.

The following option should be set to `Yes` when creating the database. This can also be modified after the DB has been created.
![image](https://user-images.githubusercontent.com/19697099/102859177-84717580-4451-11eb-9754-3379b9475f12.png)

Also the security group that you select here should allow access to port `5432`, so that your server can access it.
![image](https://user-images.githubusercontent.com/19697099/102859295-ac60d900-4451-11eb-8ff4-15f513ef6d83.png)

Let me know if this works :)

**Reply by dhuber666** [reply]
Hi I just checked it. I chose Public access and also the port is also added to the group. 
Here are screenshots that show this:

![image](https://user-images.githubusercontent.com/15130832/102904763-77b14980-4472-11eb-8434-b529929c4cdf.png)

![image](https://user-images.githubusercontent.com/15130832/102904796-8435a200-4472-11eb-862a-6bd1f37b85be.png)

Any ideas? :D 

**Edit:**
I just removed the db from AWS and started from scratch. It works now :)

**Reply by ryands17** [reply]
Does the subnet group that the database is deployed to have internet access? The Subnets would need to be associated to a Route table that has an Internet Gateway attached so that it is publicly accessible globally. Also just for testing purposes, set the access on port `5432` to allow all i.e. `0.0.0.0/0` so that it can be debugged and then you can revert it :)

Let me know if that works, as most probably that should be the issue as the rest of the params are fine.

**Reply by dhuber666** [reply]
Hiho. Thx for helping me out so much. I edited my post. Everything is working perfectly now. Maybe a small side question. How to best drop the db from AWS? Haven't found any setting other then removing the whole instance. I'm just testing a few things and this requires me to drop it many times. Do I need a seperate shell or program?

**Reply by ryands17** [reply]
One way you could drop the database is by using something like [PgAdmin](https://www.pgadmin.org/). Connect your RDS via this and drop it directly. Another way is to drop it via Prisma's [raw queries]](https://www.prisma.io/docs/concepts/components/prisma-client/raw-database-access)

```ts
await prisma.$executeRaw`drop database db-name`
```

And this should work as well.

**Reply by dhuber666** [reply]
@ryands17 thank you very much. Today suddenly the db isn't working anymore. I get the error `Can't reach database server at "dburlfromAWS" : "5432"`

The db is running on AWS normally and I didn't change a thing. Any ideas? 

pgAdmin also can't open the url from yesterday anymore. 
Status on AWS is `available`

**Reply by ryands17** [reply]
Maybe something in the security groups might have changed in this case. If pgAdmin cannot access it as well it might be the case. 

Does the Security group only allow database access from your IP address? If so, your IP address might have changed and it's inaccessible now.

**Reply by dhuber666** [reply]
Yep that was it. So it will only allow the ip I currently have when I create the db. Get it. And ofc i have a dynamic IP from my provider and it's not the same anymore. I added 0.0.0.0/0 as rule and leave it for now. Thank you very much for all your help!

**Comment by depombo** [comment]
I made this tutorial for anyone trying to get this working as of recently! https://docs.anycloudapp.com/documentation/tutorials/aws-prisma-node-psql

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that have already been marked as answered but remain open.

If this discussion still requires further input or clarification, feel free to reopen it or start a new one with updated details. Your contributions are invaluable to the community, and we’re here to help!

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and ongoing support of the Prisma community!
